src/realm/db.hpp

```diff
@@ -398,6 +398,8 @@ class DB : public std::enable_shared_from_this<DB> {
     const char* m_key;
     int m_file_format_version = 0;
     std::map<VersionID::version_type, std::weak_ptr<Transaction>> m_frozen_transactions;
+    std::shared_ptr<Transaction> m_recent_frozen_transaction;
+    VersionID::version_type m_version_of_recent_frozen_transaction;
```

Comment:
Would this ever differ from `m_recent_frozen_transaction->get_version()` (when non-null)? If not, it may be best to avoid the duplication and risk that they fall out of sync.